### PR TITLE
feat(inspect-api): support for ZoneEgress

### PR DESCRIPTION
**Pull request by lobkovilya**
### Summary

Added support for ZoneEgress

### Full changelog

* new endpoint `/zoneegresses/{name}/xds`
* new command `kumactl inspect zoneegress NAME`

### Issues resolved

Fix #XXX

### Documentation

- [ ] Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)

### Testing

- [X] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [X] Manual testing on Kubernetes

### Backwards compatibility

- [ ] Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.
- [X] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)

### CODE REVIEW

File: app/kumactl/pkg/resources/inspect_envoy_proxy_client.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	return b, nil
+}
+
+func (h *httpInspectEnvoyProxyClient) buildURL(rk core_model.ResourceKey) (*url.URL, error) {
+	var prefix string
+	if h.resDesc.Scope == core_model.ScopeMesh {
+		prefix = fmt.Sprintf("/meshes/%s", rk.Mesh)
+	}
+	plural := h.resDesc.WsPath
+	// todo(lobkovilya): rename mesh.ZoneIngressResourceTypeDescriptor.WsPath to "zoneingresses" and use it here
```
Comment (jakubdyszkiewicz) [inline app/kumactl/pkg/resources/inspect_envoy_proxy_client.go:59]: Will this be done outside of this PR? Is there a github issue?

File: pkg/xds/generator/admin_proxy_generator.go [hunk 2]
```diff
@@ -97,8 +97,7 @@ func (g AdminProxyGenerator) Generate(ctx xds_context.Context, proxy *core_xds.P
 		if err != nil {
 			return nil, err
 		}
-		// todo(lobkovilya): handle ZoneEgress
-		if (proxy.Dataplane != nil || proxy.ZoneIngress != nil) && hasCpValidationCtx {
+		if (proxy.Dataplane != nil || proxy.ZoneIngress != nil || proxy.ZoneEgressProxy != nil) && hasCpValidationCtx {
```
Comment (jakubdyszkiewicz) [inline pkg/xds/generator/admin_proxy_generator.go:100]: I mean... one of them is always not nil, right? shouldn't `hasCpValidationCtx` be enough?

**Review by jakubdyszkiewicz — APPROVED** [review]

File: pkg/xds/generator/admin_proxy_generator.go (hunk 2 above)
Comment (lobkovilya) [reply pkg/xds/generator/admin_proxy_generator.go]: Right, thanks!

File: app/kumactl/pkg/resources/inspect_envoy_proxy_client.go (hunk 1 above)
Comment (lobkovilya) [reply app/kumactl/pkg/resources/inspect_envoy_proxy_client.go:59]: Yes, I think it's covered by https://github.com/kumahq/kuma/issues/3233

**Pull request merged**